File: src/main/java/com/github/joelws/release/tracker/service/release/execution/ReadReleaseServiceExecution.java
package com.github.joelws.release.tracker.service.release.execution;


import com.github.joelws.release.tracker.dao.release.ReleaseDao;
import com.github.joelws.release.tracker.entity.release.Release;
import com.github.joelws.release.tracker.service.ServiceExecution;

public class ReadReleaseServiceExecution implements ServiceExecution<String, Release> {
    private final ReleaseDao releaseDao;

    public ReadReleaseServiceExecution(ReleaseDao releaseDao) {
        this.releaseDao = releaseDao;
    }

    @Override
    public Release execute(String identifier) {
        return releaseDao.read(identifier);
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/service/release/execution/ReadReleaseServiceExecution.kt
package com.github.joelws.release.tracker.service.release.execution


import com.github.joelws.release.tracker.dao.release.ReleaseDao
import com.github.joelws.release.tracker.entity.release.Release
import com.github.joelws.release.tracker.service.ServiceExecution
import org.apache.log4j.Logger

open class ReadReleaseServiceExecution(private val releaseDao: ReleaseDao) : ServiceExecution<String?, Release?> {

    companion object {
        private val LOGGER = Logger.getLogger(ReadReleaseServiceExecution::class.java)
    }

    override fun execute(param: String?): Release? {
        LOGGER.info("Starting ReadReleaseServiceExecution, In: $param")
        return releaseDao.read(param)

    }
}
